xcube/util/timeindex.py

```diff
@@ -0,0 +1,142 @@
+# The MIT License (MIT)
+# Copyright (c) 2022 by the xcube development team and contributors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+"""Utilities to ensure compatibility between time variables and their indexers
+
+The utilities in this module check and, where necessary, modify time indexers
+to ensure that they are compatible with the variables they are indexing.
+"""
+
+from typing import Dict, Any, Union
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+import logging
+
+logger = logging.getLogger('xcube')
+
+
+def ensure_time_label_compatible(var: Union[xr.DataArray, xr.Dataset],
+                                 labels: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
I know, there are many code locations where we assume time coordinates and dimension is named "time". But I can tell, this is wrong. We once did the same mistake for "lat" and "lon" and we still suffer from this assumption, however, most of xcube code is now more tolerant. 

```suggestion
def ensure_time_label_compatible(var: Union[xr.DataArray, xr.Dataset],
                                 labels: Dict[str, Any],
                                 time_name: str = "time") -> Dict[str, Any]:
```

Reply:
Done.